packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -105,6 +111,61 @@ class BlockListBlock extends Component {
 		return blockName;
 	}
 
+	applySelectedStyle() {
+		const {
+			isInnerBlock,
+			isNestedInnerBlock,
+			isGroupType,
+		} = this.props;
+
+		if ( isNestedInnerBlock || isGroupType ) {
+			return styles.nestedFocusedBlock;
+		}
+
+		if ( isInnerBlock ) {
+			return styles.innerBlockContainerFocused;
+		}
+
+		return styles.blockContainerFocused;
+	}
+
+	applyUnSelectedStyle() {
+		const {
+			isDashed,
+			isDimmed,
+			isNestedInnerBlock,
+			isInnerBlock,
+			isChildOfSameRootBlook,
+			isGroupType,
+			parentId,
+			isMediaTextParent,
+			getStylesFromColorScheme,
+		} = this.props;
+
+		if ( ! isDashed && isInnerBlock && ! isChildOfSameRootBlook ) {
+			return styles.blockContainerInner;
+		}
+
+		const defaultStyle = [ isDashed ? getStylesFromColorScheme( styles.blockHolderDashedBordered, styles.blockHolderDashedBorderedDark ) : styles.blockContainer ];
+
+		if ( isMediaTextParent ) {
+			return [ ...defaultStyle, styles.blockContainerMediaTextInner ];
```

Comment:
Initially `.blockContainer` which is placed in `block.native.scss` holds the style of container block. I do not affect this in this PR

I have noticed that fo MediaText block it produced an extra margin here:
<image src="https://user-images.githubusercontent.com/21242757/68314680-dd981180-00b6-11ea-8d03-df370276c294.png" width="300" />

To get this done from `MediaText` block I will need to pass style prop through:
**`InnerBlock -> BlockList -> KeyboardAwareFlatList -> BlockListBlock`** and put it to `View` holding the block.

To avoid this I decided to use `isMediaParent` directly inside `block.native.js`. 

You think it is more suitable approach ?

Reply:
I don't think the media-text block needs any external prop or context. Setting the paddings on the media container instead of the whole media-text block would avoid the double padding you saw on the inner blocks:


```jsx
const mediaContainerPadding = isSelected ? 8 : 16;
const mediaContainerSyle = {
	paddingTop: mediaContainerPadding,
	paddingLeft: shouldStack || mediaPosition === 'left' ? mediaContainerPadding : 0,
	paddingRight: shouldStack || mediaPosition === 'right' ? mediaContainerPadding : 0,
	paddingBottom: mediaContainerPadding,
	width: widthString,
};```

![media-text-padding](https://user-images.githubusercontent.com/8739/68409959-4b603e00-0188-11ea-8f77-7cbfa2d3cadd.png)

I'm not 100% sure if those are the exact paddings we want, but the point is that they would depend only on the media-text vertical (`shouldStack`) and horizontal (`mediaPosition`) layouts.

Reply:
@koke did you tried do nesting inGroup block with that solution ?

Reply:
Sorry, I don't think I tried that scenario

Reply:
Is this work in progress? @jbinda